File: DI/DI-Dagger-Java/src/main/java/model/vehiculos/Engine.java
package model.vehiculos;

import java.util.UUID;

public class Engine {

    private final String id = UUID.randomUUID().toString();

    //@Inject No lo usamos porque lo haremos en el módulo
    public Engine() {
    }

    public void start() {
        System.out.println("Engine started");
    }

    public void stop() {
        System.out.println("Engine stopped");
    }

    @Override
    public String toString() {
        return "Engine{" +
                "id='" + id + '\'' +
                '}';
    }
}

File: DI/DI-Dagger-Kotlin/src/main/kotlin/models/cars/Engine.kt
package models.cars

import java.util.*

//@Inject No lo usamos porque lo haremos en el módulo
data class Engine(val id: String = UUID.randomUUID().toString()) {

    fun start() {
        println("Engine started")
    }

    fun stop() {
        println("Engine stopped")
    }
}
